### PR TITLE
lib: Use patternfly-react Modal components to construct our dialogs

### LINKED ISSUE #13193: storage: Long dialogs are not scrollable
On the storage page, if a dialog exceeds the screen height, there's no way to scroll it.

In several other pages, such as the Virtual Machines page, it's possible to scroll the page which scrolls the dialog. This applies to desktop sizes that aren't all enough for the dialog to fit (and a scrollbar shows up) or mobile for longer dialogs (where one can scroll up and down by touch-dragging, like normal).

It's currently unknown if only the Storage page has this problem or any other part of Cockpit might also have this issue... but in a quick test of several other pages and their dialogs, I only saw the issue in the Storage page.

You can easily see the issue @ https://www.youtube.com/watch?v=DASfi39YGZ8

### PR DESCRIPTION
**Pull request by mvollmer**
This will make them scroll properly.

Not that with this change, the ".modal" DOM element is no longer a
child of the #cockpit_modal_dialog div.

Fixes #13193

### CODE REVIEW

File: pkg/lib/cockpit-components-dialog.jsx [hunk 1]
```diff
@@ -265,32 +265,20 @@ export class Dialog extends React.Component {
...
-                            <div className="modal-header">
-                                <h4 className="modal-title">{ this.props.title }</h4>
-                            </div>
-                            { this.props.body }
-                            { this.props.footer }
-                        </div>
-                    </div>
-                </div>
-            </div>
+            <Modal id={this.props.id} show animation={false}>
```
Comment (KKoukiou) [inline pkg/lib/cockpit-components-dialog.jsx]: animation={false} is probably the same like omitting it, but no need for removal. Just for clarification is that true?

Comment (mvollmer) [reply pkg/lib/cockpit-components-dialog.jsx]: No, `animation=false` is the default.

**Comment by mvollmer** [comment]
@KKoukiou, is this now using Patternfly 3?  Should we be using version 4 maybe?

**Comment by KKoukiou** [comment]
> @KKoukiou, is this now using Patternfly 3? Should we be using version 4 maybe?

PF4 modals looks quite different (buttons in footer in left side and so on) and when we port things to PF4 modals this should be done for all pages in sync.
I suggest to keep this as is.

File: pkg/lib/cockpit-components-dialog.jsx (hunk 1 above)
Comment (mvollmer) [reply pkg/lib/cockpit-components-dialog.jsx]: Dang, `animation=true` is the default.  Sorry.

**Pull request merged**